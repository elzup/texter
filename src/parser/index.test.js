import m, { parseUnit, parseInputs, parseSelects } from '.'

test('works', () => {
	expect(m('kyoko kyoko toshino.')).toMatchSnapshot('plain')
	expect(m('私は[poke:ゼニガメ|ヒトカゲ|フシギダネ]を選んだ')).toMatchSnapshot(
		'select',
	)
	expect(m('my name is (name).')).toMatchSnapshot('input')
	expect(m('[ア:a|b](a)')).toMatchSnapshot('mix')
	expect(m('[ア:a|b][イ:c|d][ウ:e|f]')).toMatchSnapshot('multiple')
	expect(
		m('before(m1){buttle:(stage)（[rule:a|b]）[result:◯|☓]}(m2)after'),
	).toMatchSnapshot('repeat')

	expect(m('hoge(match)-\\(not match) (not match\\)')).toMatchSnapshot('escape')
	expect(m('hoge{match-\\{not match} [not match\\]}')).toMatchSnapshot('escape')
})

test('parseUnit', () => {
	expect(parseInputs('[ア:a|b](a)')).toMatchSnapshot('parseInputs')
	expect(parseSelects('[ア:a|b](a)')).toMatchSnapshot('parseSelects')
	expect(parseUnit('[ア:a|b](a)')).toMatchSnapshot()
})
