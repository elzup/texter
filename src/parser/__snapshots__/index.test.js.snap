// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Object {
  "blocks": Array [
    Object {
      "text": "kyoko kyoko toshino.",
      "type": "text",
    },
  ],
  "ok": true,
}
`;

exports[`works 2`] = `
Object {
  "blocks": Array [
    Object {
      "text": "私は",
      "type": "text",
    },
    Object {
      "texts": Array [
        "ゼニガメ",
        "ヒトカゲ",
        "フシギダネ",
      ],
      "type": "select",
    },
    Object {
      "text": "を選んだ",
      "type": "text",
    },
  ],
  "ok": true,
}
`;

exports[`works 3`] = `
Object {
  "blocks": Array [
    Object {
      "text": "my name is ",
      "type": "text",
    },
    Object {
      "name": "name",
      "type": "input",
    },
    Object {
      "text": ".",
      "type": "text",
    },
  ],
  "ok": true,
}
`;

exports[`works 4`] = `
Object {
  "blocks": Array [
    Object {
      "texts": Array [
        "a",
        "b",
      ],
      "type": "select",
    },
    Object {
      "name": "a",
      "type": "input",
    },
  ],
  "ok": true,
}
`;
